controllers/contentlibrary/clustercontentlibraryitem/clustercontentlibraryitem_controller.go

```diff
@@ -260,7 +260,7 @@ func (r *Reconciler) syncImageContent(ctx goctx.Context,
 		return nil
 	}
 
-	err := r.VMProvider.SyncVirtualMachineImage(ctx, cclItem.Spec.UUID, cvmi)
+	err := r.VMProvider.SyncVirtualMachineImage(ctx, cclItem.Spec.UUID, latestVersion, cvmi)
```

Comment:
The moment we've added the 3rd param we should decide if an options struct is worth it. For example:

```go
type SyncVirtualMachineImageOpts struct {
  ContentLibraryItemID string
  LatestVersion        string
  VirtualMachineImage  client.Object
}
```

Something like that. And to be clear, the `context` should never go in an options struct.

Reply:
After a second thought, I think it might be better to make the cl/ccl-item itself as a param here so that we won't need to add more params or update the option struct if needed later.

Reply:
It doesn't matter if you need to update the option struct. The point of the option struct is to make it so the function signature doesn't have to change and break things. Put the CL/CCL item into the option struct.

Reply:
I ended up having just clitem and vmi (both in `ctrlruntime.Object` type) for this method's params. I don't expect it would change again given that all the sync info would be included in these objects.